src/prefect/infrastructure/docker.py

```diff
@@ -39,6 +40,63 @@ class ImagePullPolicy(AutoEnum):
     NEVER = AutoEnum.auto()
 
 
+class DockerRegistry(Block):
+    """
+    Connects to a Docker registry.
+
+    Requires a Docker Engine to be connectable. Login information is persisted to disk
+    at the Docker default location.
+
+    Attributes:
+        username: The username to log into the registry with.
+        password: The password to log into the registry with.
+        registry_url: The URL to the registry. Generally, "http" or "https" can be
+            omitted.
+        reauth: If already logged into the registry, should login be performed again?
+            This setting defaults to `True` to support common token authentication
+            patterns such as ECR.
+    """
+
```

Comment:
Here's a fun little Moby logo for this block
```suggestion
_logo_url = "https://images.ctfassets.net/gm98wzqotmnx/2IfXXfMq66mrzJBDFFCHTp/344dda583986d2d0db361c92dd650693/Moby-logo.webp?h=250"
```

Reply:
@madkinsz the current logo doesn't render well, we can take the one from Alex for both this and DockerContainer Block 
![image](https://user-images.githubusercontent.com/86264395/183519158-0dc06544-274b-4017-a5e2-20334bf2354d.png)